tests/tests/test_mender_gateway.py

```diff
@@ -29,6 +33,58 @@
 from testutils.infra.device import MenderDeviceGroup
 
 
+@pytest.fixture(scope="function")
+def image_with_mender_conf_and_mender_gateway_conf(request):
+    """Insert the given mender_conf into a valid_image"""
+    with tempfile.TemporaryDirectory() as d:
+
+        def cleanup():
+            shutil.rmtree(d, ignore_errors=True)
+
+        request.addfinalizer(cleanup)
+        yield lambda image, mender_conf, mender_gateway_conf: add_mender_conf_and_mender_gateway_conf(
+            d, image, mender_conf, mender_gateway_conf
+        )
+
+
+def add_mender_conf_and_mender_gateway_conf(d, image, mender_conf, mender_gateway_conf):
+    mender_conf_tmp = os.path.join(d, "mender.conf")
+    with open(mender_conf_tmp, "w") as f:
+        f.write(mender_conf)
+    mender_gateway_conf_tmp = os.path.join(d, "mender-gateway.conf")
+    with open(mender_gateway_conf_tmp, "w") as f:
+        f.write(mender_gateway_conf)
+    new_image = os.path.join(d, image)
+    shutil.copy(image, new_image)
+    instr_file = os.path.join(d, "write.instr")
+    with open(os.path.join(d, "write.instr"), "w") as f:
+        f.write(
+            """cd /etc/mender
+        rm mender.conf
+        write {local1} mender.conf
+        write {local2} mender-gateway.conf
+        """.format(
+                local1=mender_conf_tmp, local2=mender_gateway_conf_tmp,
+            )
+        )
+    subprocess.run(
+        ["debugfs", "-w", "-f", instr_file, new_image],
+        check=True,
+        stdout=subprocess.PIPE,
+    )
+    subprocess.run(
+        ["debugfs", "-R", "cat /etc/mender/mender.conf", new_image],
+        check=True,
+        stdout=subprocess.PIPE,
+    )
+    subprocess.run(
+        ["debugfs", "-R", "cat /etc/mender/mender-gateway.conf", new_image],
+        check=True,
+        stdout=subprocess.PIPE,
+    )
```

Comment:
Debug statements. Remove?

Reply:
We have them in other places... I can remove them, but I'd rather keep it consistent.

Reply:
Roger that.